src/main/java/rx/internal/util/ObjectPool.java

```diff
@@ -18,20 +18,15 @@
 package rx.internal.util;
 
 import java.util.Queue;
-import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
 
-import rx.Scheduler;
-import rx.functions.Action0;
-import rx.internal.util.unsafe.MpmcArrayQueue;
-import rx.internal.util.unsafe.UnsafeAccess;
-import rx.schedulers.Schedulers;
+import rx.internal.util.unsafe.*;
 
 public abstract class ObjectPool<T> {
     private Queue<T> pool;
     private final int maxSize;
 
-    private Scheduler.Worker schedulerWorker;
+    private final ScheduledExecutorService schedulerWorker;
```

Comment:
How does that affect memory, caches, etc to have an extra thread doing this stuff instead of just being the existing event loops?

I ask because we accidentally were running extra threads in a test with Netty+RxJava and it decreased throughput by 40%. That was far higher workload because metrics were being captured on the other threads, but it makes me less certain about choices involving putting work on other threads.